### PR TITLE
support single index read.

### PR DESCRIPTION
**Pull request by hanfei19910905**
when the sql stmt only requires index columns, we can only read index one time.
@coocood @shenli @zimulala @tiancaiamao @XuHuaiyu PTAL

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: plan/physical_plans.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ type PhysicalIndexScan struct {
 	DBName     *model.CIStr
 	Desc       bool
 	OutOfOrder bool
+	DoubleRead bool
```
Comment (shenli) [inline plan/physical_plans.go]: Add comment for this.

File: tablecodec/tablecodec.go [hunk 2]
```diff
@@ -312,11 +308,16 @@ func Unflatten(datum types.Datum, ft *types.FieldType) (types.Datum, error) {
 		return datum, nil
 	case mysql.TypeDate, mysql.TypeDatetime, mysql.TypeTimestamp:
 		var t mysql.Time
-		t.Type = ft.Tp
-		t.Fsp = ft.Decimal
-		err := t.Unmarshal(datum.GetBytes())
-		if err != nil {
-			return datum, errors.Trace(err)
+		var err error
+		if inIndex {
```
Comment (shenli) [inline tablecodec/tablecodec.go]: We should add a new request type and check if  underlying kv store support this.

File: executor/new_executor_xapi.go [hunk 3]
```diff
@@ -133,6 +134,52 @@ func (e *NewXSelectIndexExec) Close() error {
...
 // Next implements Executor Next interface.
 func (e *NewXSelectIndexExec) Next() (*Row, error) {
+	if e.indexPlan.DoubleRead {
+		return e.nextForDoubleRead()
+	}
+	return e.nextForSingleRead()
+}
+
+func (e *NewXSelectIndexExec) nextForSingleRead() (*Row, error) {
+	if e.indexPlan.LimitCount != nil && e.returnedRows >= uint64(*e.indexPlan.LimitCount) {
```
Comment (shenli) [inline executor/new_executor_xapi.go]: We can move this check to Next()

**Comment by shenli** [comment]
LGTM

**Pull request merged**